cdap-api/src/main/java/co/cask/cdap/api/schedule/DataSchedule.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+/**
+ *
+ */
+public class DataSchedule extends Schedule {
+  /**
+   * Source type, in case the {@link ScheduleType} of this schedule is {@code DATA}.
+   */
+  public enum SourceType { STREAM, DATASET }
```

Comment:
No, in the coming PRs I will make checks to be sure that the source is STREAM. You think I should remove the DATASET option until we enable dataset triggers?